hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
Well, if the proxy responds that way we have a number of problems but none of them matter that much. You see, the connection will be closed by the proxy anyway. In the absence of content-length we expect the connection to be close-delimited, but it *will* be, because that's what `proxy-connection: close` means: at the end of this response I will close the connection.

Further, such a use is largely unprecedented. Proxy-Connection was a mistake, and almost never used. A conforming, remotely-modern proxy will not issue it: instead, it will use the Connection header. This goes *doubly* true in this case when the proxy is sending a response on behalf of itself, rather than on behalf of the remote entity.

I'm inclined to say that as a 21st century HTTP implementation we should simply ignore this header field and behave as though it does not exist.